getting-started/app-services.html.md

```diff
@@ -0,0 +1,154 @@
+
+### Get the service listening on the right address within the VM
+
+Fly Proxy reaches services through a private IPv4 address on each VM, so the process should listen on `0.0.0.0:<port>` (but see [A note on IPv4 and IPv6 wildcards](#a-note-on-ipv4-and-ipv6-wildcards)).
+
+#### UDP is special
+UDP services have to listen at a different specific address, `fly-global-services`, on their VM, due to how our UDP proxying logic works.
+
+### Configure the service for Fly Proxy to reach it
+
+#### Public apps
+
+Define a [service](/docs/reference/configuration/#the-http_service-section) in `fly.toml` (or in the case of non-App-platform Machines, in individual Machine configs), making sure the `internal_port` on the service definition is the same port the process is bound to inside the VM. Without this config, Fly Proxy isn't aware of the service and can't route to it.
+
+#### Private (Flycast) apps
+
+Configure a service as for a public app, but in addition: [Flycast](/docs/reference/private-networking/#flycast-private-load-balancing) doesn't do HTTPS, so make sure you've defined a non-HTTPS service, and also ensure that `force_https = true` is **not** included in any HTTP service definition.
+
+### Allocate a public or private IP to the app
+
+#### Public apps
+
+To have Fly Proxy route requests to an app from the public internet, provision at least one public Anycast IP for the app. An app with HTTP and HTTPS services defined is allocated public IPs automatically on the first run of `fly deploy`; you can confirm with `fly ips list`.
+
+#### Private (Flycast) apps
+
+To have Fly Proxy load-balance among VMs on a non-public app, [allocate a _private_ (Flycast) IPv6 address](/docs/reference/private-networking/#assigning-a-flycast-address) using `fly ips allocate-v6 --private`, and **remove any public IPs from the app** using `fly ips release`. A Flycast IP can only be reached from within the private network it's allocated on, which must belong to the same Fly Organization as the Fly App it points to. 
+
+<section class="warning icon">
+If your configuration includes any services for Fly Proxy to route to, and the app has a public IP, that service is exposed to the whole internet.</section>
+
+### A note on Fly App IPs and Fly-Replay routing
+
+App-wide IP addresses (public and private) tell Fly Proxy which app to deliver a request to. If a service _only_ needs to be reachable by Fly Proxy in its handling of the [Fly-Replay](/docs/reference/dynamic-request-routing/) response header, its app does not need an IP; app information is incorporated directly in the header.
+
+## Private services on 6PN (direct WireGuard connections)
+
+If you don't need load balancing or other Fly Proxy features, you (or one of your apps) can connect directly on a VM's 6PN address: its address on its IPv6 private WireGuard network.
```

Comment:
```suggestion
If you don't need load balancing or other Fly Proxy features, then you (or one of your apps) can connect directly on a VM's 6PN address, which is the VM's address on its IPv6 private WireGuard network.
```
not sure about this one, the last but could also be in parentheses

Reply:
> Nice! Left some nitpicky suggestion for ya. Assuming you are using "VM" vs "Machine" here on purpose.

Yes, for now.

Reply:
I added parentheses.